aten/src/ATen/native/mps/operations/HistogramKernel.mm

```diff
@@ -305,8 +316,7 @@ void histogramdd_kernel_impl(Tensor& hist_output,
       MTLSize threadGroupSize = MTLSizeMake(tgSize, 1, 1);
       [computeEncoder dispatchThreads:gridSize threadsPerThreadgroup:threadGroupSize];
 
-      [computeEncoder endEncoding];
-      mpsStream->synchronize(SyncType::COMMIT);
```

Comment:
Can you explain why those two are removed? Because if I'm reading `endProfileKernel` correctly, it's just a no-op if profiling is not enabled, isn't it?
https://github.com/pytorch/pytorch/blob/02a7318a5ba040baca0df31fbc5d5972ae26a937/aten/src/ATen/mps/MPSProfiler.mm#L276-L280

Reply:
Yes good catch. Added an explanation in the PR description.